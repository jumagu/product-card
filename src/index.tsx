export * from './hooks';
export * from './components';
export * from './interfaces/product.interfaces';
