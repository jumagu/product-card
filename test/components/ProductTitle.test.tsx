import React from 'react';
import renderer from 'react-test-renderer';

import { product1 } from '../data/products';
import { ProductTitle, ProductCard } from '../../src/components';

describe('ProductTitle', () => {
  test('Should render correctly with the custom title', () => {
    const wrapper = renderer.create(
      <ProductTitle title="Test Product"></ProductTitle>
    );

    expect(wrapper.toJSON()).toMatchSnapshot();
  });

  test('Should render correctly with the product name', () => {
    const wrapper = renderer.create(
      <ProductCard product={product1}>
        {() => <ProductTitle></ProductTitle>}
      </ProductCard>
    );

    expect(wrapper.toJSON()).toMatchSnapshot();
  });
});
