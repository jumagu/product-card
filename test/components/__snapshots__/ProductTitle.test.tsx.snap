// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductTitle Should render correctly with the custom title 1`] = `
<span
  className="productDescription undefined"
>
  Test Product
</span>
`;

exports[`ProductTitle Should render correctly with the product name 1`] = `
<div
  className="productCard undefined"
>
  <span
    className="productDescription undefined"
  >
    Coffee Mug
  </span>
</div>
`;
